elytron/src/main/java/org/wildfly/extension/elytron/PolicyDefinitions.java

```diff
@@ -229,14 +230,22 @@ private Policy getPolicy() {
             public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
                 OperationStepHandler write = new ReloadRequiredWriteAttributeHandler(attributes) {
                     @Override
-                    protected void validateUpdatedModel(OperationContext context, Resource resource) throws OperationFailedException {
+                    protected void finishModelStage(OperationContext context, ModelNode operation, String attributeName, ModelNode newValue, ModelNode oldValue, Resource resource) throws OperationFailedException {
+                        super.finishModelStage(context, operation, attributeName, newValue, oldValue, resource);
+
                         ModelNode model = resource.getModel();
                         String defaultPolicy = context.getCurrentAddress().getLastElement().getValue();
 
                         if (model.hasDefined(ElytronDescriptionConstants.DEFAULT_POLICY)) {
                             defaultPolicy = ElytronExtension.asStringIfDefined(context, DEFAULT_POLICY, model);
                         }
 
+                        if (attributeName.equals(JACC_POLICY) || attributeName.equals(CUSTOM_POLICY)) {
+                            if (checkPolicyProviderRemoved(context, defaultPolicy, newValue, oldValue)) {
+                                throw new OperationFailedException("Cannot remove the default policy provider [ " + defaultPolicy + " ]");
```

Comment:
There's a similar problem at L177 of the existing code too. And L271, L335 and L453.